fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/PersistentStorage.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.impl;
+
+import java.time.Instant;
+
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.fcrepo.kernel.api.models.Tombstone;
+
+/**
+ * Interface for interacting with a persistent storage implementation.
+ *
+ * @author whikloj
+ * @since 2019-09-19
+ */
+public interface PersistentStorage {
+
+    /**
+     * Create a new resource on the persistent storage.
+     *
+     * @param psTxId The persistent storage transaction for this action.
+     * @param resource The new fedora resource to persist to storage.
+     * @return the resource with any new information related to the persisting..?
+     */
+    public FedoraResource create(final PersistentStorageTransaction psTxId, final FedoraResource resource);
```

Comment:
Should this method throw a PersistentStorageException?

Reply:
Out of curiosity, why is the argument name `psTxId`, rather than `psTx`.  It threw me off a little bit, and made me think that the role of `psTxId` was just to identify a transaction, rather than provide access to it.

Reply:
I'd say use runtime exceptions until it's clear what sort of checked exceptions would be needed.

Reply:
@birkland sorry I was moving fast and copying. I'll update this variable name